docs/experiments-msmarco-document.md

```diff
@@ -0,0 +1,140 @@
+# PyGaggle: Neural Ranking Baselines on [MS MARCO Document Retrieval](https://github.com/microsoft/TREC-2019-Deep-Learning)
+
+This page contains instructions for running various neural reranking baselines on the MS MARCO *document* ranking task. 
+Note that there is also a separate [MS MARCO *passage* ranking task](https://github.com/castorini/pygaggle/blob/master/docs/experiments-msmarco-passage.md).
+
+Prior to running this, we suggest looking at our first-stage [BM25 ranking instructions](https://github.com/castorini/anserini/blob/master/docs/experiments-msmarco-doc.md).
+We rerank the BM25 run files that contain ~1000 documents per query using monoT5.
+MonoT5 is pointwise reranker. This means that each document is scored independently using T5 respectively.
+
+Since it can take many hours to run these models on all of the 5193 queries from the MS MARCO dev set, we will instead use a subset of 50 queries randomly sampled from the dev set. 
+
+Note 1: Run the following instructions at root of this repo.
+Note 2: Make sure that you have access to a GPU
+Note 3: Installation must have been done from source and make sure the [anserini-eval](https://github.com/castorini/anserini-eval) submodule is pulled. 
+To do this, first clone the repository recursively.
+
+```
+git clone --recursive https://github.com/castorini/pygaggle.git
+```
+
+Then install PyGaggle using:
+
+```
+pip install pygaggle/
+```
+
+## Models
+
++ monoT5-base: Document Ranking with a Pretrained Sequence-to-Sequence Model [(Nogueira et al., 2020)](https://arxiv.org/pdf/2003.06713.pdf)
+
+## Data Prep
+
+We're first going to download the queries, qrels and run files corresponding to the MS MARCO set considered. The run file is generated by following the BM25 ranking instructions. We'll store all these files in the `data` directory.
+
+```
+wget https://www.dropbox.com/s/8lvdkgzjjctxhzy/msmarco_doc_ans_small.zip -P data
+```
+
+To confirm, `msmarco_doc_ans_small.zip` should have MD5 checksum of `aeed5902c23611e21eaa156d908c4748`.
+
+Next, we extract the contents into `data`. 
+
+```
+unzip data/msmarco_doc_ans_small.zip -d data
+```
+
+`msmarco_doc_ans_small` contains two disjoint sets, `fh` and `sh`, and each set has 25 queries. As a sanity check, we can evaluate the first-stage retrieved documents using the official MS MARCO evaluation script on first half of the dataset.
```

Comment:
```suggestion
`msmarco_doc_ans_small` contains two disjoint sets, `fh` and `sh`, and each set has 25 queries.
```
This is the eval script for MS MARCO passage and probably will break here so lets skip this